cpp/src/parquet/file_reader.cc

```diff
@@ -84,12 +87,21 @@ class SerializedRowGroup : public RowGroupReader::Contents {
  public:
   SerializedRowGroup(const std::shared_ptr<ArrowInputFile>& source,
                      FileMetaData* file_metadata, int row_group_number,
-                     const ReaderProperties& props, InternalFileDecryptor* file_decryptor)
+                     const ReaderProperties& props
+#ifdef PARQUET_ENCRYPTION
```

Comment:
This is definitely ugly. Can't we keep the classes and just set crypto variables to `nullptr` if building without encryption support?

Reply:
Let me know if you need help with this. I think we can just use `nullptr` checks to avoid invoking encryption calls if building without encryption support.

Reply:
I also see it ugly but I'm not sure how to do it better. Now I will change it as your suggestion. Thanks!